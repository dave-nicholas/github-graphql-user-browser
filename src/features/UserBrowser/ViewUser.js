import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

const ViewUser = ({ user, avatarUrl, createdAt }) => {
  return (
    <div className="card">
      <div className="card-image">
        <img src={avatarUrl} alt={user} />
      </div>
      <div className="card-stacked">
        <div className="card-content">
          User since: {moment(createdAt).format('MMMM Do YYYY')}
        </div>
      </div>
    </div>
  );
};

ViewUser.propTypes = {
  avatarUrl: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
  user: PropTypes.string.isRequired
};

export default ViewUser;
