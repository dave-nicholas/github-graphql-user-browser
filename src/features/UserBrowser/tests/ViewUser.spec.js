import React from 'react';
import { shallow } from 'enzyme';
import ViewUser from '../ViewUser';
import toJson from 'enzyme-to-json';

describe('ViewUser', () => {
  const wrapper = shallow(
    <ViewUser
      avatarUrl="http://test.com/gif"
      user="test"
      createdAt="2011-05-25T18:18:31Z"
    />
  );

  it('should render image', () => {
    expect(wrapper.contains(<img src="http://test.com/gif" alt="test" />)).toBe(
      true
    );
  });

  it('should render formatted date', () => {
    expect(wrapper.text().indexOf('May 25th 2011')).toBeGreaterThan(-1);
  });

  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
