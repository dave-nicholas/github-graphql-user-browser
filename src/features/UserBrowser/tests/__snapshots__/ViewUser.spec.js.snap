// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewUser should match snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-image"
  >
    <img
      alt="test"
      src="http://test.com/gif"
    />
  </div>
  <div
    className="card-stacked"
  >
    <div
      className="card-content"
    >
      User since: 
      May 25th 2011
    </div>
  </div>
</div>
`;
